### PR TITLE
docs: fix links

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR fixes the links broken in https://github.com/elastic/apm-server/pull/10894.

A list of broken links is here https://github.com/elastic/apm-server/pull/10891#issuecomment-1563004781.

### CODE REVIEW

**Review by basepi — APPROVED** [review]

File: docs/opentelemetry.asciidoc [hunk 1]
```diff
@@ -6,8 +6,8 @@ using the OpenTelemetry API. This allows users to utilize the Elastic APM agent'
 automatic instrumentations, while keeping custom instrumentations vendor neutral.
 
 If a span is created while there is no transaction active, it will result in an
-Elastic APM {apm-guide-ref}/transactions.html[`Transaction`]. Inner spans
-are mapped to Elastic APM {apm-guide-ref}/transaction-spans.html[`Span`].
+Elastic APM {apm-guide-ref}/data-model-transactions.html[`Transaction`]. Inner spans
+are mapped to Elastic APM {apm-guide-ref}/data-model-spans.html.html[`Span`].
```
Comment (bmorelli25) [inline docs/opentelemetry.asciidoc:10]:
```suggestion
are mapped to Elastic APM {apm-guide-ref}/data-model-spans.html[`Span`].
```

File: docs/api.asciidoc [hunk 2]
```diff
@@ -72,7 +72,7 @@ except ValueError:
  * `exc_info`: A `(type, value, traceback)` tuple as returned by https://docs.python.org/3/library/sys.html#sys.exc_info[`sys.exc_info()`]. If not provided, it will be captured automatically.
  * `date`: A `datetime.datetime` object representing the occurrence time of the error. If left empty, it defaults to `datetime.datetime.utcnow()`.
  * `context`: A dictionary with contextual information. This dictionary must follow the
-    {apm-guide-ref}/events-api.html#events-api-errors[Context] schema definition.
+    {apm-guide-ref}/api-events.html#events-api-errors[Context] schema definition.
```
Comment (bmorelli25) [inline docs/api.asciidoc:75]:
```suggestion
    {apm-guide-ref}/api-error.html[Context] schema definition.
```

**Comment by bmorelli25** [comment]
ohhhh docs ci isn't required so this auto-merged and broke the doc build

**Comment by basepi** [comment]
> ohhhh docs ci isn't required so this auto-merged and broke the doc build

I did notice it was red and assumed you were on it 😱 

We don't require docs build because it's out of our control, I just never merge if it's red unless I know it's an upstream issue or we didn't touch docs.

**Comment by bmorelli25** [comment]
> We don't require docs build because it's out of our control, I just never merge if it's red unless I know it's an upstream issue or we didn't touch docs.

Makes sense. I'll make a note not to use auto-merge in your repo for doc PRs. And yes, I'm on it!

**Pull request merged**